src/fs.cpp

```diff
@@ -154,7 +154,10 @@ std::string get_filesystem_error_message(const fs::filesystem_error& e)
 #ifdef __GLIBCXX__
 
 // reference: https://github.com/gcc-mirror/gcc/blob/gcc-7_3_0-release/libstdc%2B%2B-v3/include/std/fstream#L270
-
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wswitch"
```

Comment:
Will this be needed after the boost::fs removal?

Reply:
I think no, as this bunch of code should go: https://github.com/bitcoin/bitcoin/blob/d6e0d78c31557660274ef53cac912c468eecbe2d/src/fs.h#L66-L69